langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -919,6 +919,82 @@ function testCloneWithTypeIntArrayToUnionArray() {
     assert(messageString, "'float[]' value cannot be converted to '(lang.int:Signed16|lang.int:Unsigned8|decimal)[]'");
 }
 
+function testCloneWithTypeArrayToTuple() {
```

Comment:
can we split this test to multiple functions so that it is more readable? We can make the array  `arr` to be a global variable if needed. Also it would be better if we can move the newly created tests to a separate file as this file seems to have become really long